File: sistema-examenes-backend/sistema-examenes-backend/src/main/java/com/sistemas/examanes/servicios/ExamenService.java

package com.sistemas.examanes.servicios;

import com.sistemas.examanes.entidades.Categoria;
import com.sistemas.examanes.entidades.Examen;
import java.util.List;
import java.util.Set;


public interface ExamenService {
    
    Examen agregarExamen(Examen examen);
    Examen actualizarExamen(Examen examen);
    Set<Examen> obtenerExamenes();
    Examen obtenerExamen(Long examenId);
    void eliminarExamen(Long examenId);
    List<Examen> listarExamenesDeUnaCategoria(Categoria categoria);
    List<Examen> obtenerExamenesActivo();
    List<Examen> obtenerExamenActivosDeUnaCategoria(Categoria categoria);
    
}

File: sistema-examenes-frontend/src/app/services/examen.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ExamenService {

  constructor(private http:HttpClient) { }

  public listarCuestionarios(){
    return this.http.get(`${baseUrl}/examen/`)
  }

  public agregarExamen(examen:any){
    return this.http.post(`${baseUrl}/examen/`,examen)
  }

  public eliminarExamen(examenid:any){
    return this.http.delete(`${baseUrl}/examen/${examenid}`)
  }

  public obtenerExamen(examenId:any){
    return this.http.get(`${baseUrl}/examen/${examenId}`)
  }

  public actualizarExamen(examen:any){
    return this.http.put(`${baseUrl}/examen/`,examen)
  }

  public listarExamenPorCategoria(categoriaId:any){
    return this.http.get(`${baseUrl}/examen/categoria/${categoriaId}`)
  }

  public obtenerExamenesActivos(){
    return this.http.get(`${baseUrl}/examen/activo`)
  }

  public obtenerExamenesActivosDeUnaCategoria(categoriaId:any){
    return this.http.get(`${baseUrl}/examen/categoria/activo/${categoriaId}`)
  }
}
